File: cli/internal/graph/testdata/kotlin/basic.kt
package com.mobiai.demo

import com.x.y.Foo
import kotlinx.coroutines.flow.Flow

class LoginViewModel(private val repo: AuthRepository) {
    fun login(email: String, password: String): Flow<Result> {
        return repo.authenticate(email, password)
    }

    fun logout() {
        repo.clear()
    }
}

object Config {
    val baseUrl = "https://example.com"
}

interface AuthRepository {
    suspend fun authenticate(email: String, password: String): User?
    fun clear()
}

data class User(val id: String, val email: String)

sealed class Result {
    object Success : Result()
    data class Error(val message: String) : Result()
}

File: cli/internal/graph/testdata/swift/basic.swift
import Foundation
import Combine
import SwiftUI

class LoginViewModel: ObservableObject {
    @Published private(set) var state: LoginState = .idle
    private let repo: AuthRepository

    init(repo: AuthRepository) {
        self.repo = repo
    }

    func login(email: String, password: String) async throws -> User {
        return try await repo.authenticate(email: email, password: password)
    }

    func logout() {
        repo.clear()
    }
}

struct User: Identifiable, Codable {
    let id: String
    let email: String
}

protocol AuthRepository {
    func authenticate(email: String, password: String) async throws -> User
    func clear()
}

enum AuthError: Error {
    case invalidCredentials
    case network(underlying: Error)
}

actor SessionStore {
    private var current: User?

    func set(_ user: User) {
        current = user
    }
}

extension User {
    func displayName() -> String { email }
}
